File: plugin/src/test/java/ru/privatenull/pnauth/velocity/VelocityMessagesTest.java
package ru.privatenull.pnauth.velocity;

import net.kyori.adventure.text.Component;
import org.junit.jupiter.api.Test;
import ru.privatenull.pnauth.message.MessageFormat;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class VelocityMessagesTest {
    @Test
    void preservesJsonClickAndHoverEvents() {
        Component component = VelocityMessages.component("""
                {"text":"Open","clickEvent":{"action":"run_command","value":"/login password"},"hoverEvent":{"action":"show_text","contents":{"text":"Authenticate"}}}
                """, MessageFormat.JSON);

        assertEquals("/login password", component.clickEvent().value());
        assertNotNull(component.hoverEvent());
    }
}

File: plugin/src/test/kotlin/ru/privatenull/pnauth/velocity/VelocityMessagesTest.kt
package ru.privatenull.pnauth.velocity

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import ru.privatenull.pnauth.message.MessageFormat

class VelocityMessagesTest {
    @Test
    fun preservesJsonClickAndHoverEvents() {
        val component = VelocityMessages.component(
            """
            {"text":"Open","clickEvent":{"action":"run_command","value":"/login password"},"hoverEvent":{"action":"show_text","contents":{"text":"Authenticate"}}}
            """.trimIndent(),
            MessageFormat.JSON
        )

        assertEquals("/login password", component.clickEvent()?.value())
        assertNotNull(component.hoverEvent())
    }
}
